test/api/test_api.cpp

```diff
@@ -10,6 +10,16 @@
 using namespace duckdb;
 using namespace std;
 
+TEST_CASE("Test correlated subquery", "[api]") {
+	DuckDB db(nullptr);
+	Connection con(db);
+	con.EnableQueryVerification();
+	auto result = con.SendQuery("SELECT (0 IS DISTINCT FROM 2) = 0;");
```

Comment:
Is it necessary for this to be a C++ catch test, instead of a regular sqllogictest?

Reply:
I don't think so, this was how the issue was brought up. I'll change it to a sqllogictest